megatron/devutil.py

```diff
@@ -0,0 +1,36 @@
+import time
+
+class Metric:
+    """
+    Dumb utility to collect and report average wall-time metrics.
+    """
+    def __init__(self, label):
+        self.label = label
+        self.measurements = []
+    def collect(self, measurement):
+        self.measurements.append(measurement)
+    def report(self):
+        print(self.label, len(self.measurements), sum(self.measurements) / len(self.measurements))
+
+def monitor_method_wall_times(metric, obj, methodname):
+    """
+    Measure timings for a method on an object.
+
+    For instance:
+
+    >>> metric = Metric('!LNORM')
+    >>> monitor_method_wall_times(metric, layernorm, 'forward')
+    """
+    oldmeth = obj[methodname]
+
+    def newmeth(*args, **kw):
+        start = time.time()
+        try:
+            return oldmeth(*args, **kw)
+        finally:
+            end = time.time()
```

Comment:
Using time.time() is inaccurate for CUDA code, especially distributed code. Use torch cuda events like in https://github.com/EleutherAI/cookbook/tree/main/benchmarks/communication